event_routing_backends/processors/xapi/constants.py

```diff
@@ -14,6 +14,7 @@
 XAPI_VERB_FAILED = 'http://adlnet.gov/expapi/verbs/failed'
 XAPI_VERB_EXPERIENCED = 'http://adlnet.gov/expapi/verbs/experienced'
 XAPI_VERB_NAVIGATED = 'https://w3id.org/xapi/dod-isd/verbs/navigated'
+XAPI_VERB_CHANGED = 'https://w3id.org/xapi/dod-isd/verbs/changed'
```

Comment:
Looking at this in context I think we may just want to re-use the `registered` statement again. At is stands a learner can register and unregister any number of times and the course mode is passed in, so we're going to have to deduplicate those rows and find the most recent enrollment mode. Syntactically I think that's better than a "user changed course" verb which seems more like a course publish event.

@ziafazal do you or anyone from the original authors have an opinion on the best type of statement to transform the course enrollment mode event to?

Reply:
@bmtcril Cool yes -- that would make this change easier, but more importantly, makes the data we need easier for us to use.

@rpenido is on a limited timeline for this work -- is it ok if he proceeds with this change along the lines you've laid out?

Reply:
@rpenido Hmm.. From your https://github.com/openedx/event-routing-backends/issues/292#issuecomment-1573081240, it looks like the tracking event JSON contains only the course mode we're changing _from_, not the course mode we're changing _to_.  Am I reading your comment correctly?

If that's the case, we may need to add data to the tracking log on the edx-platform side. Because if we log a `registered` event using the _from_ mode, then that will throw us off when trying to assess "what is the most recent active enrollment mode for a given user+course?"

Reply:
@pomegranited I agree with @bmtcril we can use `registered` verb in this case too. Also we don't need to create a new transformer we can use [existing transformer ](https://github.com/openedx/event-routing-backends/blob/37a081450495a0bcb862fccb28d10ed37ff257a1/event_routing_backends/processors/xapi/event_transformers/enrollment_events.py#L53) by adding a line like this at the top of it.
```
@XApiTransformersRegistry.register('edx.course.enrollment.mode_changed')
```

Reply:
Done!